lib/rules/indent.js

```diff
@@ -856,6 +856,10 @@ module.exports = {
                         previousElementLastToken.loc.end.line - countTrailingLinebreaks(previousElementLastToken.value) > startToken.loc.end.line
                     ) {
                         offsets.setDesiredOffsets(element.range, firstTokenOfPreviousElement, 0);
+
+                        sourceCode.getCommentsBefore(element).forEach(comment => {
+                            offsets.setDesiredOffsets(comment.range, firstTokenOfPreviousElement, 0);
```

Comment:
Could this be combined with the line above into something like this instead?

```js
offsets.setDesiredOffsets(
    [previousElement.range[1], element.range[1]],
    firstTokenOfPreviousElement,
    0
);
```

`offsets.setDesiredOffsets` accepts a text range as the first argument, and modifies the indentation of all tokens in that range, without needing to iterate over all of those tokens. So it's not necessary to call it multiple times on separate consecutive tokens, and ends up being slightly worse for performance (although it's unlikely to matter much in this case since users probably don't have large numbers of consecutive comments between elements).